invisible_cities/cities/detsim.py

```diff
@@ -0,0 +1,216 @@
+import numpy  as np
+import tables as tb
+import pandas as pd
+import os
+from functools import partial
+
+from .. core import system_of_units as units
+from .. reco import tbl_functions   as tbl
+
+from .. database import load_db  as db
+
+from . components import city
+from . components import print_every
+from . components import copy_mc_info
+from . components import collect
+from . components import calculate_and_save_buffers
+from .. dataflow  import dataflow   as fl
+
+from .. io.rwf_io           import rwf_writer
+from .. io.run_and_event_io import run_and_event_writer
+from .. io. event_filter_io import event_filter_writer
+
+from .  components                import MC_hits_from_files
+from .. detsim.simulate_electrons import generate_ionization_electrons
+from .. detsim.simulate_electrons import drift_electrons
+from .. detsim.simulate_electrons import diffuse_electrons
+from .. detsim.light_tables_c     import LT_SiPM
+from .. detsim.light_tables_c     import LT_PMT
+from .. detsim.s2_waveforms_c     import create_wfs
+from .. detsim.detsim_waveforms   import s1_waveforms_creator
+
+
+def hits_selector (active_only: bool=True):
+    """
+    Filtering function that selects hits. (see :active_only: description)
+
+    Parameters:
+        :active_only: bool
+            if True, returns hits in ACTIVE
+            if False, returns hits in ACTIVE and BUFFER
+    Returns:
+        :select_hits: Callable
+            function that select the hits depending on :active_only: parameter
+    """
+    def select_hits(x, y, z, energy, time, label):
+        sel = (label == "ACTIVE")
+        if not active_only:
+            sel =  sel | (label == "BUFFER")
+        return x[sel], y[sel], z[sel], energy[sel], time[sel], label[sel]
+    return select_hits
+
+
+def ielectron_simulator(wi: float, fano_factor: float, lifetime: float,
+                        transverse_diffusion: float, longitudinal_diffusion: float, drift_velocity:float,
+                        el_gain: float, el_gain_sigma: float):
+    """
+    Function that simulates electron creation, drift, diffusion and photon generation at the EL
+
+    Parameters: floats
+        parameter names are self-descriptive.
+    Returns:
+        :simulate_ielectrons:
+            function that returns the positions emission times and number of photons at the EL
+    """
+    def simulate_ielectrons(x, y, z, time, energy):
+        electrons = generate_ionization_electrons(energy, wi, fano_factor)
```

Comment:
How about `num_electrons`, or `nelectrons` so it is clear you are getting the number of electrons?